linkerd/app/outbound/src/lib.rs

```diff
@@ -32,7 +32,7 @@ use std::{
     time::Duration,
 };
 
-mod discover;
+pub mod discover;
```

Comment:
I'd prefer that we reexport the things we need than reexport the module.

```suggestion
mod discover;
```